pkg/ingester/ingester.go

```diff
@@ -3062,6 +3066,31 @@ func (i *Ingester) ModeHandler(w http.ResponseWriter, r *http.Request) {
 	_, _ = w.Write([]byte(respMsg))
 }
 
+func (i *Ingester) getInstanceLimits() *InstanceLimits {
+	// Don't apply any limits while starting. We especially don't want to apply series in memory limit while replaying WAL.
+	if i.State() == services.Starting {
+		return nil
+	}
+
+	if i.cfg.InstanceLimitsFn == nil {
+		return defaultInstanceLimits
+	}
+
+	l := i.cfg.InstanceLimitsFn()
+	if l == nil {
+		return defaultInstanceLimits
+	}
+
+	return l
+}
+
+// stopIncomingRequests is called during the shutdown process.
+func (i *Ingester) stopIncomingRequests() {
+	i.stoppedMtx.Lock()
+	defer i.stoppedMtx.Unlock()
+	i.stopped = true
+}
+
```

Comment:
No changes on these functions.. I just moved to be close to the other ingesters functions.